src/SonarScanner.MSBuild.Common/AnalysisProperties/AggregatePropertiesProvider.cs

```diff
@@ -63,6 +58,25 @@ public IEnumerable<Property> GetAllProperties()
         return allProperties;
     }
 
+    public IDictionary<PropertyProviderKind, IList<Property>> GetAllPropertiesPerProvider()
```

Comment:
What we need is the `GetAllProperties()` implementation here. You should:
* Make another overload of TryGetProperty: `public bool TryGetProperty(string key, out Property property, out IAnalysisPropertyProvider provider)` and copy the code from the existing TryGetProperty
* The existing TryGetProperty should delegate to the new overload and just ignore the `provider` out argument.
* Create `public IEnumerable<(Property, IAnalysisPropertyProvider)> GetAllPropertiesWithProvider()`. and copy the code from `GetAllProperties`
* Replace the `TryGetProperty` call with the new overload and return the tuple
* Change `GetAllProperties()` to call `GetAllPropertiesWithProvider()` and Select the property from the tuple.